lib/hpke/src/userinfo_vc.cpp

```diff
@@ -298,13 +300,15 @@ UserInfoClaimsAddress::from_json(const nlohmann::json& address_json)
 /// UserInfoClaims
 ///
 UserInfoClaims
-UserInfoClaims::from_json(const nlohmann::json& cred_subject_json)
+UserInfoClaims::from_json(const std::string& cred_subject)
 {
+  const auto& cred_subject_json = nlohmann::json::parse(cred_subject);
+
   std::optional<UserInfoClaimsAddress> address_opt = {};
 
   if (cred_subject_json.contains(address_attr)) {
-    address_opt =
-      UserInfoClaimsAddress::from_json(cred_subject_json.at(address_attr));
+    address_opt = UserInfoClaimsAddress::from_json(
+      cred_subject_json.at(address_attr).dump());
```

Comment:
I don't love this back and forth between strings and JSON.  Maybe there's some way to have static functions that take `json`?  So that they're hidden from callers but we can still use them.

Reply:
This is the difficult thing about this library. If we have any use of our json library in the public api's then the consumer will have to build and link to our json library. If they use another library then they would have to convert. I think you understand that. I will think about the static function that might do the conversion. Out only options in the signature of a function is string or bytes. what do you have in mind?

Reply:
What I had in mind was something like:

```
// Standard from_json method so that `json::get` works
static void from_json(const json& j, UserInfoClaimsAddress& addr) {
  // current body of UserInfoClaimsAddress::from_json
}

static void from_json(const json& j, UserInfoClaims& addr) { 
  // current body of UserInfoClaims::from_json
  // including using get_optional<UserInfoClaimsAddress>(...)
}

// String methods just call through to JSON methods
UserInfoClaims
UserInfoClaims::from_json(const std::string& cred_subject) {
  return json::parse(cred_subject).get<UserInfoClaims>();
}

UserInfoClaimsAddress
UserInfoClaimsAddress::from_json(const std::string& address) {
  return json::parse(address).get<UserInfoClaimsAddress>();
}
```